middleware/gzip/filter.go

```diff
@@ -72,7 +84,7 @@ func DefaultMIMEFilter() MIMEFilter {
 // matches any of the registered ones. It returns true if
 // found and false otherwise.
 func (m MIMEFilter) ShouldCompress(r *http.Request) bool {
-	return m.Types.Contains(r.Header.Get("Content-Type"))
+	return m.Types.Contains(r.Header.Get("Accept"))
```

Comment:
Okay.

The only remaining issue I have with using the client's Accept header is that it doesn't really know what kind of resource is being served - only that it expects a certain type. Plus only browsers are going to send it and only for some types... Chrome sends image/webp for nearly any type of image, as well as */* in the second position. So I don't think it really gains us that much.